packages/unchained-client/src/evm/ethereum/parser/cowswap.ts

```diff
@@ -0,0 +1,26 @@
+import { Tx } from '../../../generated/ethereum'
+import { Dex, TradeType, TxParser } from '../../../types'
+import { SubParser, txInteractsWithContract, TxSpecific } from '../../parser'
+import { COWSWAP_CONTRACT_MAINNET } from './constants'
+
+export class Parser implements SubParser<Tx> {
+  async parse(tx: Tx): Promise<TxSpecific | undefined> {
+    if (!txInteractsWithContract(tx, COWSWAP_CONTRACT_MAINNET)) return
+    if (!(tx.tokenTransfers && tx.tokenTransfers.length)) return
+
+    const trade = {
+      dexName: Dex.CowSwap,
+      type: TradeType.Trade
```

Comment:
This is wrongly taken from the ZRX and Thor parser implementations.

We should either consolidate the `TradeType` enum, or use the `chain-adapters` one. Trade type Txs all fall back to the default case in web.